controllers/platformoperator_controller.go

```diff
@@ -102,10 +109,51 @@ func (r *PlatformOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	u := updater.NewPlatformOperatorUpdater(r.Client)
+	defer func() {
+		if err := u.Apply(ctx, po); err != nil {
+			log.Error(err, "failed to update status")
+		}
+	}()
+
+	u.UpdateStatus(
+		updater.SetPhase(platformopenshiftiov1alpha1.PhaseFailing),
+		updater.EnsureCondition(metav1.Condition{
+			Type:    platformopenshiftiov1alpha1.TypeInstalled,
+			Status:  metav1.ConditionUnknown,
+			Reason:  platformopenshiftiov1alpha1.ReasonInstalling,
+			Message: "",
+		}),
+	)
+
+	rc, err := handleCatalogConnection(catalogSourceAddress)
+	if err != nil {
+		log.Error(err, "failed to connect to the requisite catalog")
```

Comment:
nit: might be nice to log the `catalogSourceAddress` here as well?

Reply:
Definitely.